week-2/mandatory/3-api/cyf_ecommerce_api/server.js

```diff
@@ -0,0 +1,215 @@
+        host:'localhost',
+        database:'cyf_ecommerce',
+        password:'hedyeh1365',
+        port:'5432'
+    }
+);
+
+function capitalizeFirstLetter(str) {
+    return str.charAt(0).toUpperCase() + str.slice(1);
+  }
+
+
+app.get("/customers",function(req,res){
+    db.query("select * from customers",function (err,result){
+        if(err==undefined){
+            console.log(result)
+            res.status(200).json({result})
+        }
+    })
+})
+
+app.get("/customers/:id", (req,res)=>{
+    const custId=parseInt(req.params.id)
+    db.query("SELECT * FROM customers WHERE id = $1", [custId],(err,result)=>{
+        if (err==undefined){
+            res.status(200).json({customer:result.rows[0]})
+        }
+    })
+})
+// Add a new POST endpoint `/customers` to create a new customer with name, address, city and country.
+app.post("/customers",(req,res)=>{
+const NewName=req.body.name
+const NewAddress=req.body.address
+const NewCity=req.body.city
+const NewCountry=req.body.country
+const query="INSERT INTO customers (name, address, city, country) " +
+"VALUES ($1, $2, $3, $4)";
+db.query(query,[NewName,NewAddress,NewCity,NewCountry],(err,result) => {
+    if(err==undefined)
+    res.send("Customer created.");
+    else{
+        res.status(500).json({error: err});
+    }
+  })
+})
+
+
+
+app.get("/suppliers",function(req,res){
+    db.query("select * from customers",function (err,result){
+        if(err==undefined){
+            console.log(result)
+        }
+    })
+})
+// Update the previous GET endpoint `/products` to filter the list of products by name using a query parameter, for example `/products?name=Cup`. This endpoint should still work even if you don't use the `name` query parameter!
+
+app.get('/products', function (req, res) {
+    const name = (req.query.name);
+    if (name) {
+      db.query(
+        "select p.product_name, a.unit_price, s.supplier_name from"+
+        " products p join product_availability a on(p.id=a.prod_id)"+
+        " join suppliers s on (s.id=a.supp_id)"+
+         " where product_name like '%'|| $1 || '%'",
+        [capitalizeFirstLetter(name)],
+        (error, result) => {
+          if (error == undefined){
+            res.status(200).send({products:result.rows});
+          }
+        }
+      );
+    } else {
+      db.query("select p.product_name, a.unit_price, s.supplier_name from"+
+      " products p join product_availability a on(p.id=a.prod_id)"+
+      " join suppliers s on (s.id=a.supp_id)", (error, result) => {
+        if (error == undefined) {
+            res.status(200).json({products:result.rows})
+        }
+      });
+    }
+  });
+
+//Add a new POST endpoint `/products` to create a new product.
+app.post("/products",(req,res)=>{
+    const NewProduct=req.body.product_name
+   const query="INSERT INTO products (product_name) VALUES ($1)"
+        db.query(query,[NewProduct],(err)=>{
+        if(err==undefined){
+            res.send("Product created."); 
```

Comment:
As a good practice, I recommend to send the newly created entity back in the response or at least the ID of the entity. So that the API caller will know what has been created, or what ID the request has been saved with so they can then retrieve.